FWCore/ParameterSet/python/Types.py

```diff
@@ -40,11 +40,27 @@ def __init__(self,type):
         super(_ProxyParameter,self).__init__()
         self.__dict__["_ProxyParameter__type"] = type
         self.__dict__["_ProxyParameter__value"] = None
+        if hasattr(self.__type,"_default") and self.__type._default is not None:
+            self.setValue(self.__type._default)
     def setValue(self, value):
         v = self.__type(value)
         if not _ParameterTypeBase.isTracked(self):
             v = untracked(v)
         self.__dict__["_ProxyParameter__value"] = v
+    def _returnTypeWithValue(self, valueWithType):
+        if isinstance(valueWithType, type(self)):
+            return valueWithType
+        if isinstance(self.__type, type):
```

Comment:
Where does the variable `type` come from? (or does it refer to the `type()` function?)

Reply:
`type` isn't actually a function, it is actually the base class for all `classes`. E.g.
```python3 
>>> import FWCore.ParameterSet.Config as cms
>>> type(cms.int32)
<class 'type'>
>>> if isinstance(cms.int32, type): print('True')
... 
True
```

Reply:
Ok